### PR TITLE
feat: Cleanup data roadworks

### PR DESCRIPTION
**Pull request by alainajaffer**

### CODE REVIEW

File: packages/ref-data-uploaders/cleanup-roadworks/ospoint.d.ts
Comment (DeannaSharma) [inline packages/ref-data-uploaders/cleanup-roadworks/ospoint.d.ts:1]: Is this file needed? It doesn't look like we use it for our cleanup roadworks function.

File: stacks/S3.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { BucketEncryption } from "aws-cdk-lib/aws-s3";
 import { Bucket, Stack, StackContext, toCdkDuration } from "sst/constructs";
+import { Topic } from "aws-cdk-lib/aws-sns";
```
Comment (DeannaSharma) [inline stacks/S3.ts]: Sorry I think I pushed this up as my SST stack was being weird earlier. Could you delete this line please?

File: stacks/Uploaders.ts [hunk 2]
```diff
@@ -111,4 +113,36 @@ export function UploadersStack({ stack }: StackContext) {
...
 
     nptgBucketCdk.addEventNotification(EventType.OBJECT_CREATED, new LambdaDestination(nptgUploader));
+
+    const cleanupRoadworks = new Function(stack, `ref-data-service-cleanup-roadworks`, {
+        bind: [cluster],
+        functionName: `ref-data-service-cleanup-roadworks-${stack.stage}`,
+        handler: "packages/ref-data-uploaders/cleanup-roadworks/index.main",
+        runtime: "nodejs18.x",
+        timeout: 600,
+        memorySize: 3008,
```
Comment (DeannaSharma) [inline stacks/Uploaders.ts]: Can we drop memorySize to 1024?

3008 seems like a lot - think we should start low and bump up if needs be to avoid excessive cost.

File: packages/ref-data-uploaders/cleanup-roadworks/ospoint.d.ts
Comment (alainajaffer) [reply packages/ref-data-uploaders/cleanup-roadworks/ospoint.d.ts:1]: done

File: stacks/Uploaders.ts (hunk 2 above)
Comment (alainajaffer) [reply stacks/Uploaders.ts]: done

File: stacks/S3.ts (hunk 1 above)
Comment (alainajaffer) [reply stacks/S3.ts]: done

**Comment by alainajaffer** [comment]
One roadwork where the workStatus was "works in progress" but the permitStatus was "revoked" - I think we need to ask the street manager team what this actually means in practice. As from the data we are not sure if it means roadworks would still be actually happening on the road or not.

File: packages/ref-data-uploaders/cleanup-roadworks/index.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import { Database, getDbClient, waitForDb } from "@reference-data-service/core/db";
+import { randomUUID } from "crypto";
+import * as logger from "lambda-log";
+import { Kysely, sql } from "kysely";
+
+export const deleteOldRoadworks = async (dbClient: Kysely<Database>) => {
+    await dbClient
+        .deleteFrom("roadworks")
+        .where(sql`DATE_ADD(DATE_FORMAT(left(actualEndDateTime,10),'%Y-%m-%d'), INTERVAL 7 DAY)`, "<", sql`CURDATE()`)
+        .where((query) =>
```
Comment (DeannaSharma) [inline packages/ref-data-uploaders/cleanup-roadworks/index.ts]: This query isn't quite right, as it is it won't delete revoked or refused permit unless that have an actualEndDateTime (looks at the preprod DB the actualEndDateTime is usually null)

It's a bit complicated but the query should look like this:
```
 await dbClient
        .deleteFrom("roadworks")
        .where((qb) =>
            qb.or([
                qb.and([
                    qb(
                        sql`DATE_ADD(DATE_FORMAT(left(actualEndDateTime,10),'%Y-%m-%d'), INTERVAL 7 DAY)`,
                        "<",
                        sql`CURDATE()`,
                    ),
                    qb("permitStatus", "=", "closed"),
                ]),
                qb.and([
                    qb(
                        sql`DATE_ADD(DATE_FORMAT(left(lastUpdatedDateTime,10),'%Y-%m-%d'), INTERVAL 7 DAY)`,
                        "<",
                        sql`CURDATE()`,
                    ),
                    qb.or([
                        qb("permitStatus", "=", "revoked"),
                        qb("permitStatus", "=", "refused"),
                        qb("permitStatus", "=", "cancelled"),
                    ]),
                ]),
            ]),
        )
        .execute();
        ```
        
        I've also added the cancel permitStatus

Comment (alainajaffer) [reply packages/ref-data-uploaders/cleanup-roadworks/index.ts]: done

**Review by DeannaSharma — APPROVED** [review]
